crates/biome_service/src/workspace_watcher.rs

```diff
@@ -86,6 +89,9 @@ pub struct WorkspaceWatcher {
     //       `Send`, but it appears all platform implementations are.
     watcher: Box<dyn Watcher + Send>,
 
+    /// Tracks the folders we are watching, including the scan kind per folder.
+    watched_folders: FxHashMap<PathBuf, ScanKind>,
```

Comment:
I wonder if we can ask for this information from the `Workspace`, instead of copying this information for all folders. The thing is, there's **one** `ScanKind` per project, and that one instance is computed from the configuration. I've noticed that we also have a method that takes the greater `ScanKind` if we've different ones, but this isn't always true.

We already have the logic that computes `ScanKind` inside the `find_project_root`, we just need to expose it and we should be able to give it to the watcher on request.

Reply:
There is a very slight duplication between the workspace and the watcher indeed, but I do think this is where it belongs. Note we don't copy this information for all folders, only for the _watched_ folder, so in a typical setup we still store it only once per project.

I think however, that in order to address #6838, we may want to implement more granular watching and track it right here. But that's none of the workspace's business.

So the division as I see it is: the workspace determines _what_ to watch, and the watcher is responsible for the actual watching.

Reply:
> but I do think this is where it belongs.

Can you explain why? IMHO, it's worth documenting. At the moment, I disagree, and if there's a specific reason to duplicate the information, let's at least explain it in a comment so we don't lose the insight.

Reply:
Can you explain why you disagree? I just spent three paragraphs explaining my reasoning, so I'm unsure which part you're disagreeing with.